server/battles/BattleFlowProcessor.cpp

```diff
@@ -757,8 +759,11 @@ void BattleFlowProcessor::stackTurnTrigger(const CBattleInfoCallback & battle, c
 				});
 				spells::BattleCast parameters(&battle, st, spells::Mode::ENCHANTER, spell);
 				parameters.setSpellLevel(bonus->val);
-				parameters.massive = true;
-				parameters.smart = true;
+				//todo: not hardcode
+				if (spellID != SpellID::ARMAGEDDON) {
+					parameters.massive = true;
+					parameters.smart = true;
+				}
```

Comment:
This is not a fix but ugly workaround.
What if mod adds another spell similar to Armageddon? 
Are you sure that Armageddon is the only such spell, what about others? Like Death Ripple or possible some non-damaging spells.
What's so special in Armageddon that requires hardcoding it?

Reply:
so check the the range of the spell, if a damage spell which range is X, and smart is false, not set its smart and massive mark to true?